ExampleMod/ExamplePlayer.cs

```diff
@@ -634,6 +635,103 @@ public class ExamplePlayer : ModPlayer
+			}
+
+			// The texture we want to display on our player
+			Texture2D texture = mod.GetTexture("Items/Armor/ExampleBreastplate_" + gender + "Body_Glowmask");
+
+			// The following variables (until drawData) are all copied 1:1 from vanilla code that handles breastplate drawing.
+			// It is advised to always copy the vanilla code that handles a particular EquipType, if you wish to make a glowmask for it,
+			// so it seamlessly overlaps
+			float drawX = (int)drawInfo.position.X + drawPlayer.width / 2;
+			float drawY = (int)drawInfo.position.Y + drawPlayer.height - drawPlayer.bodyFrame.Height / 2 + 4f;
+
+			// The rotation origin of the texture
+			// Because we are drawing a breastplate and it has the same alignment as the players' body, we use bodyOrigin
+			Vector2 origin = drawInfo.bodyOrigin;
+
+			// The position this texture will be drawn at on the screen
+			// Because we are drawing a breastplate and it has the same alignment as the players' body, we use bodyPosition
+			Vector2 position = new Vector2(drawX, drawY) + drawPlayer.bodyPosition - Main.screenPosition;
+
+			// The alpha of the texture when the player is blinking (immune)
+			// Should stay like that
+			float alpha = (255 - drawPlayer.immuneAlpha) / 255f;
+
+			// The color of the glowmask, pick what you want
+			Color color = Color.White;
+
+			// The frame that gets cut out from our texture that then will be drawn
+			// Because we are drawing a breastplate and it has the same alignment as the players' body, we use bodyFrame
+			Rectangle frame = drawPlayer.bodyFrame;
+
+			// The texture rotation around the origin
+			// Because we are drawing a breastplate and it's tied to players' body, we use bodyRotation
+			float rotation = drawPlayer.bodyRotation;
+
+			// The horizontal/vertical flipping (based on direction, gravity)
+			// Should stay like that
+			SpriteEffects spriteEffects = drawInfo.spriteEffects;
+
+			DrawData drawData = new DrawData(texture, position, frame, color * alpha, rotation, origin, 1f, spriteEffects, 0);
+
+			// The shader that's applied to the texture
+			// Depending on what EquipType this glowmask is representing, you need to assign the proper shader to it
+			// (This references the dye)
+			// Because this is a breastplate, we want to use the dye that is in the breastplate slot
+			drawData.shader = drawInfo.bodyArmorShader;
+
+			// Finally we register our glowmask so it will be drawn later by the game
+			Main.playerDrawData.Add(drawData);
+
+			// Here we generate visual dust while our glowmask is visible
+			if (Main.rand.NextBool(10)) {
+				int index = Dust.NewDust(drawPlayer.position, drawPlayer.width, drawPlayer.height, 135);
+				Dust dust = Main.dust[index];
+				dust.noGravity = true;
+				dust.noLight = true;
+				dust.fadeIn = Main.rand.NextFloat(0.5f, 0.8f);
+
+				// Same deal as before with drawData.shader
+				dust.shader = GameShaders.Armor.GetSecondaryShader(drawInfo.bodyArmorShader, drawPlayer);
```

Comment:
Very cool, please add: `// Gives the dust the same shader effect that is equipped for the armor`